docs/resource/README.md

```diff
@@ -102,7 +102,117 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair
+(in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique).
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+
+If the instance has no inherent unique ID that can be used as the value of this attribute,
+implementations MAY to generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. When enough data is available,
+implementations SHOULD use Version 5 and MUST use the following UUID as the namespace:
+`4d63009a-8d0f-11ee-aad7-4c796ed8e320`.
+
+UUIDs are typically recommended, as only an opaque yet reproducible value for
+the purposes of identifying a service instance is needed. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file,
+the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the input MUST be prefixed with
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}`, followed by the
+the workload identifier, which should tentatively stable. The workload identifier is the final piece of
+information making the source of telemetry to be uniquely idenfiable. For instance, it might be the `container.id`
+or `container.name` in containerized or Kubernetes environments. This means that the same service yields the
+same UUID if the same identifier (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+Users are encouraged to explicitly set the `service.instance.id` themselves, given they are in the best position
+to determine what's a service instance. For instance, in some cases, all containers in a given pod might be part
+of the same service, while in other cases, each container would independently generate their own telemetry. When
+it's not possible or desirable to explicitly generate a `service.instance.id`, users are encouraged to set set
+enough information via their workload automation, such as Helm. In that case, SDKs would have enough information to
```

Comment:
```suggestion
enough information via their workload automation, such as Helm. In that case, SDKs would have enough data to
```
Nit: I'd see this as data, and the `service.instance.id` as the actual information.